File: FrontEnd/src/types/quiz/quizJobDTO.ts
export interface QuizJobDTO {
  jobId: string;
  userId: string;
  noteTitle: string;
  status: "queued" | "processing" | "success" | "failed";
  timestamp: string;
  quizId?: string;
  errorMessage?: string;
  isViewed?: boolean;
  createdAt: number;
}

File: BackEnd/StudyNest.Common/Models/DTOs/EntityDTO/Quizzes/QuizJobDTO.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace StudyNest.Common.Models.DTOs.EntityDTO.Quizzes
{
    public class QuizJobDTO
    {
        public string JobId { get; set; }
        public string UserId { get; set; }
        public string NoteTitle { get; set; }
        public string Status { get; set; }
        public string? QuizId { get; set; }
        public string? ErrorMessage { get; set; }
        public string Timestamp { get; set; }
        public bool IsViewed { get; set; }
        public DateTime CreatedAt { get; set; }
    }
}
